src/avt/Queries/Pick/avtLocateCellQuery.C

```diff
@@ -167,8 +173,10 @@ avtLocateCellQuery::Execute(vtkDataSet *ds, const int dom)
     int foundCell = -1;
 
     // Find the cell, intersection point, and distance along the ray.
-    //
-    if (ds->GetDataObjectType() != VTK_RECTILINEAR_GRID)
+    // Don't use the RectilinearGrid fast-path if there are ghosts, as it
+    // takes longer to remove the ghosts than to use the fast-path.
+    if ( ds->GetDataObjectType() != VTK_RECTILINEAR_GRID ||
+         dataAtts.GetContainsGhostZones())
```

Comment:
So, just to validate the logic here...any mesh block that is either NOT rectilinear or HAS ghost zones will wind up going through the slower logic to find the picked cell instead of using the potentially faster `RGridFindCell()` logic, right?

In practice, in any parallel decomposed mesh, even one with all rectilinear blocks, every block is likely going to have *some* ghost zones (well, maybe just ghost *nodes* is the common case where neighboring blocks abut) and so almost exclusively now, we'll always pick the slower path.

I know your modification comment in the preamble mentions something about performance. Do you have a good feel for impact on our large multi-block, structured gridded codes? Maybe those aren't typically truly rectilinear anyways.

Reply:
I am not sure about the impact for real-sized data. I just did some timing comparisons with our test data.
I decided to look into changing the ghost test to GetContainsExternalBoundaryGhosts (which satisfies the users dataset).  Am currently evaluating the regression result differences for said change.

Reply:
@markcmiller86 Changing the test to GetContainsExternalBoundaryGhosts reduces the amount of times pick will not take the rectilinear grid fast-path. (at least for the pick tests in our repo).   I think this should be sufficient, what do you think?

Reply:
Sounds good. I say we ship it :shipit: